src/util/threadpool.rs

```diff
@@ -280,9 +275,22 @@ impl<Q: ScheduleQueue<T>, T: Debug> TaskPool<Q, T> {
     }
 
     fn pop_task(&mut self) -> Option<Task<T>> {
+        if let Some(task) = self.task_queue.pop() {
+            return Some(task);
+        }
+        // try fill queue when queue is empty.
+        self.try_fill_queue();
         self.task_queue.pop()
     }
 
+    fn try_fill_queue(&mut self) {
+        while let Ok(mut task) = self.jobs.try_recv() {
```

Comment:
If we add a limitation for task's number in `threadpool`, it could also limit the max count  in `channel` here ? @siddontang